core/src/subgraph/runner.rs

```diff
@@ -1041,7 +1096,22 @@ where
 
         self.revert_state(subgraph_ptr.number)?;
 
-        Ok(Action::Continue)
+        // If any datasource has reached its end block in the range [revert_to_ptr, subgraph_ptr]
+        // then we need to restart the blockstream to reset the TriggerFilter.
+        let needs_restart: bool = self
+            .inputs
+            .end_blocks
+            .range(revert_to_ptr.number..=subgraph_ptr.number)
+            .next()
+            .is_some();
+
+        let action = if needs_restart {
+            Action::Restart
+        } else {
+            Action::Continue
+        };
+
+        Ok(action)
     }
```

Comment:
This the part where we determine if the blockstream is to be restarted on a reorg. If there is atleast one datasource which reached its endBlock during the block range for which the reorg is happening then we need to restart